Fix WPT failure for `css/cssom/shorthand-values.html`

iidmsa opened the pull request:
#### 4bcc6b54119d25022bbd333e24d3f2aa8b64e54a
<pre>
Fix WPT failure for `css/cssom/shorthand-values.html`
<a href="https://bugs.webkit.org/show_bug.cgi?id=299999">https://bugs.webkit.org/show_bug.cgi?id=299999</a>
<a href="https://rdar.apple.com/162244690">rdar://162244690</a>

Reviewed by Tim Nguyen.

This patch aligns WebKit with Gecko / Firefox and Blink / Chromium.

This patch fixes an issue in serialization where in the past, we used to not serialize `border` property
and were not canonical. Now it aligns with other browsers and returns canonical reference.

CSS spec: <a href="https://drafts.csswg.org/cssom/#serialize-a-css-declaration-block">https://drafts.csswg.org/cssom/#serialize-a-css-declaration-block</a>

* LayoutTests/fast/css/remove-shorthand-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-sizing/stretch/block-height-008-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/shorthand-values-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/css/css-sizing/stretch/block-height-008-expected.txt:
* Source/WebCore/css/StyleProperties.cpp:
(WebCore::canUseShorthandForLonghand):

Canonical link: <a href="https://commits.webkit.org/303844@main">https://commits.webkit.org/303844@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/33895896742080dbad84eacc8609395187bf90ed

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/133661 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/6166 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/44837 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/141232 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/85716 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/135531 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/6689 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/6033 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/102237 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/69589 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/136608 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/4726 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/119823 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/83037 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/4605 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/2222 "Passed tests") | [⏳ 🛠 wpe-cairo ](https://ews-build.webkit.org/#/builders/WPE-Cairo-Build-EWS "Waiting in queue, processing has not started yet") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/113741 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/37936 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/143880 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/5838 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/38517 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/110621 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/5921 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/4993 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/110805 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/28119 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/4456 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/116075 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/59587 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/5892 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/34386 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/5739 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/69361 "Built successfully") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/5983 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/5847 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

karlcow reviewed (changes requested):
You will need to fix the expected file for this test.
https://github.com/WebKit/WebKit/blob/c91b99ca61167b1a157693f669c69fe0a68b571e/LayoutTests/imported/w3c/web-platform-tests/css/cssom/shorthand-values-expected.txt#L10


And check why the two other tests are failing.

karlcow reviewed (commented):
Thanks for the change!

iidmsa committed:
Fix WPT failure for `css/cssom/shorthand-values.html`
https://bugs.webkit.org/show_bug.cgi?id=299999
rdar://162244690

Reviewed by Tim Nguyen.

This patch aligns WebKit with Gecko / Firefox and Blink / Chromium.

This patch fixes an issue in serialization where in the past, we used to not serialize `border` property
and were not canonical. Now it aligns with other browsers and returns canonical reference.

CSS spec: https://drafts.csswg.org/cssom/#serialize-a-css-declaration-block

* LayoutTests/fast/css/remove-shorthand-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-sizing/stretch/block-height-008-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/shorthand-values-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/css/css-sizing/stretch/block-height-008-expected.txt:
* Source/WebCore/css/StyleProperties.cpp:
(WebCore::canUseShorthandForLonghand):

Canonical link: https://commits.webkit.org/303844@main
